toolkits/kaldi/gigaspeech_data_prep.sh

```diff
@@ -1,47 +1,58 @@
 #!/usr/bin/env bash
 # Copyright 2021  Xiaomi Corporation (Author: Yongqing Wang)
+#                 Seasalt AI, Inc (Author: Guoguo Chen)
 
 set -e
 
-if [ $# != 1 ]; then
-  echo "Usage: "
-  echo "  $0 <data-dir>"
+if [ $# -lt 1 ] || [ $# -gt 2 ]; then
+  echo "Usage: $0 <data-dir> [<subset-prefix>]"
+  echo " e.g.: $0 data/ gigaspeech"
   exit 1
 fi
 
 data_dir=$1
+prefix=
+if [ $# -eq 2 ]; then
+  prefix=${2}_
+fi
 
 stage=1
 
 meta_dir=$GIGA_SPEECH_LOCAL_ROOT/data/meta
-declare -A dic_sets
-dic_sets=([train]="XL" [dev]="DEV" [test]="TEST")
+declare -A subsets
+subsets=([train]="XL" [dev]="DEV" [test]="TEST")
 
 if [ $stage -le 1 ]; then
-  # all corpus
-  [ ! -d $data_dir/corpus ] && mkdir -p $data_dir/corpus
+  # All data.
+  [ ! -d $data_dir/${prefix}corpus ] && mkdir -p $data_dir/${prefix}corpus
 
   for f in utt2spk wav.scp text segments utt2dur reco2dur; do
-    [ -f $meta_dir/$f ] && cp $meta_dir/$f $data_dir/corpus/
+    [ -f $meta_dir/$f ] && cp $meta_dir/$f $data_dir/${prefix}corpus/
   done
 
-  utt2spk=$data_dir/corpus/utt2spk
-  spk2utt=$data_dir/corpus/spk2utt
-  utt2spk_to_spk2utt.pl <$utt2spk >$spk2utt || (echo "Error: utt2spk to spk2utt" && exit 1)
+  utt2spk=$data_dir/${prefix}corpus/utt2spk
+  spk2utt=$data_dir/${prefix}corpus/spk2utt
+  utt2spk_to_spk2utt.pl <$utt2spk >$spk2utt ||\
+    (echo "$0: Error: utt2spk to spk2utt" && exit 1)
 
-# Delete <*> tag
-  sed -i '/<MUSIC>/d' $data_dir/corpus/text
-  sed -i '/<NOISE>/d' $data_dir/corpus/text
-  sed -i "s|<[^>]*>||g" $data_dir/corpus/text
-  sed -i 's/[ ][ ]*/ /g' $data_dir/corpus/text
+  # Delete <*> tag
+  sed -i '/<MUSIC>/d' $data_dir/${prefix}corpus/text || exit 1
```

Comment:
@wangyongqing0731  I think we should explicitly deal with ALL existing tags in GigaSpeech, which I listed here: https://github.com/SpeechColab/GigaSpeech/wiki , only dealing with <MUSIC> and <NOISE> while skipping other tags(<SIL> <OTHER>) seems incomplete to me.